web/src/lib.rs

```diff
@@ -435,127 +435,8 @@ impl Ruffle {
-                        "pointerdown",
-                        player_mouse_down_callback.as_ref().unchecked_ref(),
-                    )
-                    .warn_on_error();
-            }
-            if let Some(window_mouse_down_callback) = instance.window_mouse_down_callback.take() {
-                instance
-                    .window
-                    .remove_event_listener_with_callback_and_bool(
-                        "pointerdown",
-                        window_mouse_down_callback.as_ref().unchecked_ref(),
-                        true,
-                    )
-                    .warn_on_error();
-            }
-            if let Some(mouse_up_callback) = instance.mouse_up_callback.take() {
-                instance
-                    .canvas
-                    .remove_event_listener_with_callback(
-                        "pointerup",
-                        mouse_up_callback.as_ref().unchecked_ref(),
-                    )
-                    .warn_on_error();
-            }
-            if let Some(mouse_wheel_callback) = instance.mouse_wheel_callback.take() {
-                instance
-                    .canvas
-                    .remove_event_listener_with_callback(
-                        "wheel",
-                        mouse_wheel_callback.as_ref().unchecked_ref(),
-                    )
-                    .warn_on_error();
-            }
-            if let Some(key_down_callback) = instance.key_down_callback.take() {
-                instance
-                    .window
-                    .remove_event_listener_with_callback(
-                        "keydown",
-                        key_down_callback.as_ref().unchecked_ref(),
-                    )
-                    .warn_on_error();
-            }
-            if let Some(key_up_callback) = instance.key_up_callback.take() {
-                instance
-                    .window
-                    .remove_event_listener_with_callback(
-                        "keyup",
-                        key_up_callback.as_ref().unchecked_ref(),
-                    )
-                    .warn_on_error();
-            }
-            if let Some(unload_callback) = instance.unload_callback.take() {
-                instance
-                    .window
-                    .remove_event_listener_with_callback(
-                        "unload",
-                        unload_callback.as_ref().unchecked_ref(),
-                    )
-                    .warn_on_error();
-            }
-
-            // Cancel the animation handler, if it's still active.
-            if let Some(id) = instance.animation_handler_id {
-                instance
-                    .window
-                    .cancel_animation_frame(id.into())
-                    .warn_on_error();
-            }
-        }
-
-        // Player is dropped at this point.
+        let _ = self.remove_instance();
+        // Instance is dropped at this point.
```

Comment:
(Nit/question: I do not know Rust, but, like for RAII in C++, is `drop()` called once the scope here is exited? If so, that is rather neat).

Reply:
Exactly. You can learn more about `Drop` [here](https://doc.rust-lang.org/std/ops/trait.Drop.html) and [here](https://doc.rust-lang.org/rust-by-example/trait/drop.html).